src/libsync/propagateupload.cpp

```diff
@@ -381,7 +383,12 @@ qint64 UploadDevice::readData(char *data, qint64 maxlen)
         }
         _bandwidthQuota -= maxlen;
     }
-    std::memcpy(data, _data.data() + _read, maxlen);
+
+    auto c = _file.read(data, maxlen);
+    if (c != maxlen) {
```

Comment:
should we not just do `_read += c; return c;` ?

Reply:
@ogoffart If `c >= 0` yeah. I mimiced the previous code here but that should be unnecessary.